### PR TITLE
FindByKey and FindByUsername should not include deleted accounts.

### PR DESCRIPTION
**Pull request by cristinamanum**
The default behavior for FindByKey and FindByUsername should not include deleted accounts.

### CODE REVIEW

File: src/NuGetGallery/Services/UserService.cs [hunk 1]
```diff
@@ -352,17 +352,33 @@ public virtual IList<User> FindByUnconfirmedEmailAddress(string unconfirmedEmail
             }
         }
 
-        public virtual User FindByUsername(string username)
+        public virtual User FindByUsername(string username, bool includeDeleteRecord = false)
```
Comment (chenriksson) [inline src/NuGetGallery/Services/UserService.cs]: Arg names should match the interface (plural). Also, would rename to simply `includeDeleted` as `Record` is kind of generic data term that we don't use elsewhere in the gallery.

File: src/NuGetGallery/Services/UserService.cs [hunk 2]
```diff
@@ -352,17 +352,33 @@ public virtual IList<User> FindByUnconfirmedEmailAddress(string unconfirmedEmail
...
+                return UserRepository.GetAll()
+                .Include(u => u.Roles)
+                .Include(u => u.Credentials)
+                .SingleOrDefault(u => u.Username == username);
+            }
             return UserRepository.GetAll()
+                .Where(u => !u.IsDeleted)
                 .Include(u => u.Roles)
                 .Include(u => u.Credentials)
                 .SingleOrDefault(u => u.Username == username);
```
Comment (chenriksson) [inline src/NuGetGallery/Services/UserService.cs]: nit: can reuse code by wrapping only Where clause in the condition.

```
var users = UserRepository.GetAll();
if (includeDeleted)
{
    users = users..Where(u => u.IsDeleted);
}
return users.Include(...).Include(...).SingleOrDefault(...);
```

Comment (chenriksson) [reply src/NuGetGallery/Services/UserService.cs]: Same for below. Also, username should be case insensitive search to match our DB collation.

Comment (shishirx34) [reply src/NuGetGallery/Services/UserService.cs]: this equality comparison is case insensitive by default due to db collation, why would it matter if username has any casing?

File: src/NuGetGallery/Services/UserService.cs (hunk 1 above)
Comment (cristinamanum) [reply src/NuGetGallery/Services/UserService.cs]: Updated

File: src/NuGetGallery/Services/UserService.cs (hunk 2 above)
Comment (cristinamanum) [reply src/NuGetGallery/Services/UserService.cs]: Updated (first comment) , second comment @shishirx34 is correct.

**Pull request merged**